### DISCUSSION TITLE
Adding an optional argument to `sep.extract` to return the "thresh_map"

### DISCUSSION THREAD

**Discussion by mapfiable** [sep-developers/sep#100] [Ideas]
In the same vein as there is an option to to have [`sep.extract`](https://sep.readthedocs.io/en/v1.0.x/api/sep.extract.html) return a segmap of all the pixels that costitute detections, it would be nice to also have the option to have it return a map of all the pixels above the threshold. Going by the documentation, this map is calculated easily enough manually (e.g. something like "`thresh_map = data > thresh*err`"). But since it is already calculated once, it would be convenient to not having to determine it again.

**Comment by kbarbary** [comment]
Would this be equivalent to `segmap > 0`? (e.g., True if pixel is part of an object, otherwise False)

**Reply by mapfiable** [reply]
I'm not entirely sure, but I think not. As I understood it, the `segmap` is the map of all the pixels that belong to detections, correct? But at the same time, not necessarily all pixels that are above the threshold will eventually belong to a detection, yes? So in that case, the `thresh_map` would genreally be different from the `segmap` and only be the same if all pixels above the threshold do end up being part of a detection.

Also, until I just now read the [other post](https://github.com/kbarbary/sep/discussions/103) about `sep.sum_` I always thought that "pixels that belong to a detection" would include any pixel within the aperture, not just the ones above the threshold. So that was also confusing to me. So in that sense, maybe it would also be nice to have an "extended" `segmap` or whatever you want to call it, that includes all the pixeles that are inside the respective apertures of the respective detections, and not just the ones above the threshold. 

**EDIT** 

On the same note, it would also be nice to already have the fully integrated flux over all pixels within the detection apertures (`sep.sum_`) in the array that is retured by `sep.extract`. (Maybe also once for the convolved and unconvolved area.)
